pkg/networkservice/connectioncontext/dnscontext/client.go

```diff
@@ -66,6 +72,29 @@ func NewClient(options ...DNSOption) networkservice.NetworkServiceClient {
 }
 
 func (c *dnsContextClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
+	if request.Connection == nil {
+		request.Connection = &networkservice.Connection{}
+	}
+	if request.Connection.GetContext() == nil {
+		request.Connection.Context = &networkservice.ConnectionContext{
+			DnsContext: &networkservice.DNSContext{},
+		}
+	}
+	if request.Connection.GetContext().GetDnsContext() == nil {
+		request.Connection.Context.DnsContext = &networkservice.DNSContext{}
+	}
+
+	var initialClientDNSConfigs []*networkservice.DNSConfig
+	if v, ok := metadata.Map(ctx, true).Load(dnsContextClientRefreshKey); ok {
+		initialClientDNSConfigs = v.([]*networkservice.DNSConfig)
+	} else {
+		initialClientDNSConfigs = request.Connection.Context.DnsContext.Configs
+		metadata.Map(ctx, true).Store(dnsContextClientRefreshKey, initialClientDNSConfigs)
+	}
```

Comment:
Seems like we could a bit simplify this

```suggestion 
	var initialClientDNSConfigs, _ = metadata.Map(ctx, true).LoadOrStore(dnsContextClientRefreshKey, request.Connection.Context.DnsContext.Configs)
```